### PR TITLE
chore(test): move npm registries to separate servers and to the `tests/registry` folder

### PR DESCRIPTION
**Pull request by dsherret**
1. Moves the npm registries to their own dedicated ports.
2. Moves the data files out of `tests/testdata/npm/registry` to `tests/registry/npm`.

### CODE REVIEW

**Review by bartlomieju — APPROVED** [review]

File: tests/util/server/src/lib.rs [hunk 1]
```diff
@@ -304,6 +304,8 @@ async fn get_tcp_listener_stream(
   futures::stream::select_all(listeners)
 }
 
+pub const TEST_SERVERS_COUNT: usize = 28;
```
Comment (bartlomieju) [inline tests/util/server/src/lib.rs:307]: Can we somehow compute this number so we don't find ourselves chasing it again in a few months?

File: tests/util/server/src/servers/mod.rs [hunk 2]
```diff
@@ -122,60 +122,56 @@ pub async fn run_all_servers() {
...
+    h2_only_server_fut.boxed_local(),
+    h2_grpc_server_fut.boxed_local(),
+    registry_server_fut.boxed_local(),
+    provenance_mock_server_fut.boxed_local(),
+  ];
+  futures.extend(npm_registry_server_futs);
+  futures.extend(private_npm_registry_1_server_futs);
+
+  // update this constant when not equal
+  assert_eq!(futures.len(), TEST_SERVERS_COUNT);
```
Comment (bartlomieju) [inline tests/util/server/src/servers/mod.rs:172]: This is good enough 👍 skip the previous comment

File: tests/util/server/src/servers/npm_registry.rs [hunk 3]
```diff
@@ -4,51 +4,83 @@ use crate::npm;
...
+  S: Future<Output = HandlerOutput> + 'static,
+{
   let npm_registry_addr = SocketAddr::from(([127, 0, 0, 1], port));
+  let ipv6_loopback = Ipv6Addr::new(0, 0, 0, 0, 0, 0, 0, 1);
+  let npm_registry_ipv6_addr =
+    SocketAddr::V6(SocketAddrV6::new(ipv6_loopback, port, 0, 0));
+  vec![
+    run_npm_server_for_addr(npm_registry_addr, error_msg, handler)
+      .boxed_local(),
+    // necessary because the npm binary will sometimes resolve localhost to ::1
```
Comment (bartlomieju) [inline tests/util/server/src/servers/npm_registry.rs:61]: This comment should be moved above?

Comment (dsherret) [reply tests/util/server/src/servers/npm_registry.rs:61]: It's just for the ipv6 one right below

**Pull request merged**